### PR TITLE
Convert mod_related_items to new structure

### PR DESCRIPTION
**Pull request by carlitorweb**
### Summary of Changes
This PR converts mod_related_items to the new structure

### Testing Instructions
- Use Joomla 4.3
- Create an instance of the module Articles Related. Note how it works.
- Apply patch
- Delete administrator/cache/autoload_psr4.php
- Refresh the page, make sure the module still displays the same as before.

### Link to documentations
Please select:
- [x] No documentation changes for docs.joomla.org needed

- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: modules/mod_related_items/src/Helper/RelatedItemsHelper.php [hunk 1]
```diff
@@ -25,146 +32,210 @@
 /**
  * Helper for mod_related_items
  *
- * @since  1.5
+ * @since  __DEPLOY_VERSION__
```
Comment (laoneo) [inline modules/mod_related_items/src/Helper/RelatedItemsHelper.php]: Revert

File: modules/mod_related_items/src/Helper/RelatedItemsHelper.php [hunk 2]
```diff
@@ -25,146 +32,210 @@
...
-    public static function getList(&$params)
+    public function __construct(array $config = [])
     {
-        $db        = Factory::getDbo();
-        $app       = Factory::getApplication();
-        $input     = $app->getInput();
-        $groups    = Factory::getUser()->getAuthorisedViewLevels();
-        $maximum   = (int) $params->get('maximum', 5);
-        $factory   = $app->bootComponent('com_content')->getMVCFactory();
+        $this->input = array_key_exists('input', $config) ? $config['input'] : Factory::getApplication()->getInput();
```
Comment (laoneo) [inline modules/mod_related_items/src/Helper/RelatedItemsHelper.php]: Why is this needed, you have the input from the app in the getRelatedArticles function.

File: modules/mod_related_items/src/Helper/RelatedItemsHelper.php [hunk 3]
```diff
@@ -25,146 +32,210 @@
...
+                    $db->quoteName('a.publish_down') . ' IS NULL',
+                    $db->quoteName('a.publish_down') . ' >= :nowDate2',
+                ],
+                'OR'
+            )
+            ->bind([':nowDate1', ':nowDate2'], $now);
+
+        // Filter by language
+        if ($props->app->getLanguageFilter()) {
+            $query->whereIn($db->quoteName('a.language'), [Factory::getApplication()->getLanguage()->getTag(), '*'], ParameterType::STRING);
```
Comment (laoneo) [inline modules/mod_related_items/src/Helper/RelatedItemsHelper.php]:
```suggestion
            $query->whereIn($db->quoteName('a.language'), [$props->app->getLanguage()->getTag(), '*'], ParameterType::STRING);
```

File: modules/mod_related_items/tmpl/default.php [hunk 4]
```diff
@@ -13,15 +13,21 @@
 use Joomla\CMS\HTML\HTMLHelper;
 use Joomla\CMS\Language\Text;
 
+if (!count($list)) {
```
Comment (laoneo) [inline modules/mod_related_items/tmpl/default.php]: Please revert them as well, even when it is only intendation as it will show up then on every site which has an override that something has changed.

Comment (carlitorweb) [reply modules/mod_related_items/tmpl/default.php]: And where use it? If `$list` is empty where we check this and return? In the Dispatcher?

**Comment by carlitorweb** [comment]
@laoneo I revert the Helper in this one.

**Comment by carlitorweb** [comment]
@laoneo here in the test, I created a menu item for show a single article view, and from there get the related articles. 
I added to the `db_createMenuItem` the option `link: 'index.php?option=com_content&view=article&id=1` where `id=1` is the ID of the main article to view. **Is correct this ID?** For example, in my local site, I needed add `id=15` because I had already some articles there.

**Comment by carlitorweb** [comment]
Nevermind now, I just featured the article, click on it, and get the related articles from there. So not matter what ID he will have, and by default joomla come with the featured menu item.

**Comment by carlitorweb** [comment]
I will include the cache

**Comment by laoneo** [comment]
Please wait. @joomdonation will have a look tomorrow and when all works, then you can update. Otherwise you copy my mistakes.

**Comment by carlitorweb** [comment]
Okay. I already tested in the category module, work fine. Mainly when need load in the article view.

File: modules/mod_related_items/tmpl/default.php (hunk 4 above)
Comment (laoneo) [reply modules/mod_related_items/tmpl/default.php]: Please do it in another pr against the 5.0 branch as it counts as new feature, as the root tag will then not be displayed anymore.

**Comment by carlitorweb** [comment]
@laoneo done

**Comment by laoneo** [comment]
Cool, now we need some tests.

**Comment by drmenzelit** [comment]
@carlitorweb can you update your branch?

**Comment by laoneo** [comment]
Thanks!
